src/python/pants/jvm/jdk_rules.py

```diff
@@ -45,6 +47,19 @@ def append_only_caches(self) -> dict[str, str]:
         return self.coursier.append_only_caches
 
 
+VERSION_REGEX = re.compile(r"version \"(.+?)\"")
+
+
+def parse_jre_major_version(version_lines: str) -> int | None:
+    for line in version_lines.splitlines():
+        m = VERSION_REGEX.search(line)
+        if m:
+            pos = m[1].find(".")
```

Comment:
This could be better written as 

```
major_version, dot, remainder = m[1].partition(".")
return int(major_version)
```

In this, `major_version` will have everything up to the dot if it exists, otherwise the full string.